salt/metalk8s/addons/prometheus-operator/deployed/prometheus-rules.sls

```diff
@@ -281,4 +281,52 @@ spec:
       for: 15m
       labels:
         severity: critical
+  - name: kubernetes-apps
+    rules:
+    - alert: KubeContainerOOMKilled
+      annotations:
+        description: 'Container {{ $labels.namespace }}/{{ $labels.pod }}/{{ $labels.container }}
+          in Node {{ $labels.node }} caused Pod restart by have been exceed its memory allocation 
+          limit (OOMKilled). It accumulates a total of {{ $value }} restarts.'
+        summary: 'Pod killed by exceeding memory allocation limits (OOMKilled).'
+      expr: |-
+        (
+          (
+            max_over_time(
+              kube_pod_container_status_last_terminated_reason{
+                job="kube-state-metrics",
+                namespace=~".*",
+                reason="OOMKilled"
+              }[5m]
+            )
+          )
+          * on (namespace, pod, container) kube_pod_container_status_restarts_total
+          * on (namespace, pod) group_left(node) kube_pod_info
+        ) > 0
+      for: 1s
+      labels:
+        severity: warning
+    - alert: KubeContainerOOMKillSurge
+      annotations:
+        description: 'OOMKill Surge: Container {{ $labels.namespace }}/{{ $labels.pod }}/{{ $labels.container }}
+          in Node {{ $labels.node }} caused Pod restart by have been exceed its memory allocation 
+          limit (OOMKilled). It accumulates {{ $value }} restarts in last 5 minutes.'
```

Comment:
Here it's not "$value restarts" but more "$value restarts because of OOM kill, I think it's important to mention.

Btw I wonder if we want this other rule, just counting the number of OOM kill restarts is enough to me, because a pod may restart for good reason, so catching the total number of restarts for an alert is not necessarily interesting (even if we filter on OOMKilled at the beginning, it means in the alert message you will have the total number of the restart when maybe all of them are "legit" except the OOMKilled one)

I would suggest going only with a warning alert when there is an OOMKilled and to do so I think we do not need the `kube_pod_container_status_last_terminated_reason` metrics, we can handle it directly with `container_oom_events_total`

Reply:
As explained in this comment ( https://scality.atlassian.net/browse/ARTESCA-4300?focusedCommentId=346858 ) the metric `container_oom_events_total` is capable to indicates OOMKill events for a period of 5 minutes.
My proposition ( https://scality.atlassian.net/browse/ARTESCA-4300?focusedCommentId=346867 ) is to raise 2 distinct alerts : a CRITICAL one, based on `container_oom_events_total` to indicate OOMKill surge and a WARNING that will stay active still concerned pods restarts.